modyn/config/schema/pipeline/trigger/drift/evidently.py

```diff
@@ -5,39 +5,52 @@
 from modyn.config.schema.pipeline.trigger.drift.metric import BaseMetric
 
 
-class _EvidentlyBaseDriftMetric(BaseMetric):
+class _EvidentlyPcaDriftMetric(BaseMetric):
     num_pca_component: int | None = Field(None)
 
 
-class EvidentlyModelDriftMetric(_EvidentlyBaseDriftMetric):
+class EvidentlyModelDriftMetric(_EvidentlyPcaDriftMetric):
     id: Literal["EvidentlyModelDriftMetric"] = Field("EvidentlyModelDriftMetric")
     bootstrap: bool = Field(False)
-    quantile_probability: float = 0.95
-    threshold: float = Field(0.55)
+    quantile_probability: float = Field(
+        0.95, description="Threshold for the evidently api, unused in modyn. Usage discourged."
+    )
+    threshold: float = Field(0.55, description="Threshold for the evidently api, unused in modyn. Usage discourged.")
 
 
-class EvidentlyRatioDriftMetric(_EvidentlyBaseDriftMetric):
+class EvidentlyRatioDriftMetric(_EvidentlyPcaDriftMetric):
     id: Literal["EvidentlyRatioDriftMetric"] = Field("EvidentlyRatioDriftMetric")
     component_stattest: str = Field(
         "wasserstein",
         description="The statistical test used to compare the components.",
     )
-    component_stattest_threshold: float = Field(0.1)
-    threshold: float = Field(0.2)
+    component_stattest_threshold: float = Field(
+        0.1, description="Threshold for the evidently api, unused in modyn. Usage discourged."
+    )
+    threshold: float = Field(0.2, description="Threshold for the evidently api, unused in modyn. Usage discourged.")
 
 
-class EvidentlySimpleDistanceDriftMetric(_EvidentlyBaseDriftMetric):
+class EvidentlySimpleDistanceDriftMetric(_EvidentlyPcaDriftMetric):
```

Comment:
I don't get yet why everything is a Pca drift metric from an inheritance sense. I suppose if num_pca_components is none, no PCA is applied?

Reply:
Renamed to mixin, it's just for avoiding have to duplicate the field everywhere. And yes, none is no pca. We don't use that in our experiments though.